OpenMage/magento-lts
# Enhancement for categories management and products grids

### Description

It allow to keep current selected tab on store switch, and to keep current category and current store on refresh in categories management. It prevents to add a new category without a parent category.

It add columns to products grid in categories (status, visibility, action).

![image](https://user-images.githubusercontent.com/31816829/137367890-e10e76db-0887-4af0-85b0-31353271e268.png)

It open product edit in a popin (without header and footer) with the current selected store in categories and in associated products of configurable products. You can also open the edit page in a new tab or in a new window.

![image](https://user-images.githubusercontent.com/31816829/137368143-df771ced-da72-4acf-8a88-2d5153ab86a9.png)

It allow to check or uncheck products only when clicking on the checkbox or in the td that contain the checkbox for all grids (products in categories ; related, upsell, crosssell, associated (for grouped and configurable products)).

OpenMage 20.0.13 / PHP 7.4.11 - 8.0.11

### Manual testing scenarios

**TEST 1**
Go to Catalog / Categories.

- Try to create a new subcategory without selecting a parent. This is not possible.
- You can check or uncheck products only when clicking on td input.
- New columns "Status", "Visibility" and "Action".
- The edit link opens product edit page in popup (with the current store).
- You can open the [Edit] link in a new browser tab (middle click or right click "Open in a new tab/window").
- Use Name, Price, Status and Visibility values from selected store.

**TEST 2**
Go to Catalog > Categories the select any category.
Go to Category Products tab.

- Select another store with store switcher.  The current tab is not lost.
- Press F5 or refresh the page. The current category and the current store are not lost.
- Uncheck all products then reset filter or filter or sort or page change. After AJAX all products are unchecked.


**TEST 3**
Go to Catalog > Products and choose any configurable product. For Associated Products grid.

- You can check or uncheck products only when clicking on td input.
- The [Edit] link opens product edit page in popup (with the current store)
- You can open the [Edit] link on a new tab (middle click or right click open in a new tab/window)

**TEST 4**
Go to Catalog > Products and choose any product.
For Related, Up-sells and Cross-sells, Associated (for grouped products) grids.

- Uncheck all products then reset filter or filter or sort or page change. After AJAX all products are unchecked.
- Check some products then reset filter or filter or sort or page change. After AJAX selected products are checked.
- You can check check or uncheck products only when clicking on td input.

### Contribution checklist
